### PR TITLE
add code coverage

### PR DESCRIPTION
**Pull request by smlng**
This PR addresses issue #62 and integrates http://codecov.io into travis, see for example: https://codecov.io/github/smlng/rtrlib/rtrlib?ref=9d52582ef95a1eb491c475c95cbbf005b2a3e028

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,10 @@ script:
     - make
     - make test
     - make clean
-    - cmake -D CMAKE_BUILD_TYPE=Release .
+    - cmake -D CMAKE_BUILD_TYPE=Release -DENABLE_COVERAGE=On -ENABLE_COVERAGE_ALL=on .
```
Comment (waehlisch) [inline .travis.yml]: Does `-ENABLE_COVERAGE_ALL=on` not need a `-D`?

**Comment by smlng** [comment]
I followed configuration for cmake from [here](https://github.com/RWTH-ELP/CMake-codecov). The flag seems to be unused (or not required) anyway, so I removed it from the travis config.

**Comment by waehlisch** [comment]
OK. Can you squash the commits? Rest looks fine. 

Btw, I don't know how useful this tool really is but we should give it a try.

**Comment by smlng** [comment]
Squash done! I think it could be quite helpful as it shows which code parts are not covered by any of our tests. So we might be able to improve that ...

**Pull request merged**